### PR TITLE
Implement the latest API proposals for windowing, among other fixes

### LINKED ISSUE #80: Multiple calls to GetInput extension lead to new glfw input contexts
# Summary
Multiple calls to the GetInput extension will create a new glfw input context each time. Therefore event registering on other than the first input context will not raise the events.

# Steps to reproduce
- Platform: Any
- Framework Version: Any
1. Create one method for initialize keyboard events and one for initialize mouse events.
2. In each method use GetInput() to retrieve the input context and add event handlers for keyboard events (first method) and mouse events (second method).
3. Mouse event callbacks are not called as the used input context is not considered by Silk. Only the first one is.

The problem is the method GetInput below. It creates a new instance of the input context instead of reusing the existing one if present.

```csharp
    public class GlfwInputPlatform : IInputPlatform
    {
        public bool IsApplicable(IWindow window)
        {
            return window is GlfwWindow;
        }

        public IInputContext GetInput(IWindow window)
        {
            // the cast isn't needed, but is used to make sure that what we've got is actually a GlfwWindow
            return new GlfwInputContext((GlfwWindow)window); // <- Problem, should reuse context when called for the second time
        }
    }
```

# Possible solutions
- Reuse context on second and subsequent calls
- Documentation that each time it is called a new context is created (in this case rename to `CreateInput`)
- Make Silk register each of the created contexts and ensure that all of them are handled

I would prefer first fix.

### LINKED ISSUE #83: Monitor API

### LINKED ISSUE #84: Window & View separation
- [ ] I have definitely checked that there are no proposals for this already in the [proposals folder](https://github.com/Ultz/Silk.NET-Docs)

# Summary of feature
Currently, our windowing API has a lot of APIs that aren't really applicable on mobile platforms and after trying to write a UWP windowing backend, I figured that this needs a bit of work. I propose that we separate the core APIs that are common among desktop and mobile platforms into their own set of interfaces, prefixed by `IView`. `IWindow` will then inherit the `IView` interfaces so that the APIs aren't duplicated and to maximise ease of cross-platform support.

Also, a `GetOrCreateView` method will have to be added as mobile window usually only support one view (for UWP this is `CoreWindow`, for Android this is `ANativeWindow`, etc etc)

I'll add a proposal shortly detailing this.

# Comments
https://discordapp.com/channels/521092042781229087/587346162802229298/658816946703171585
High Priority - Required for an internal project

### PR DESCRIPTION
**Pull request by Perksey**
# Summary of the PR
Implements the latest API proposals, as well as fixes a bug or implements an enhancement or two.

# Related issues, Discord discussions, or proposals
Closes #84 
Closes #83 
Closes #80 

# Further Comments

### CODE REVIEW

**Comment by Pyrdacor** [comment]
How is the state of this PR?

**Comment by Perksey** [comment]
Currently on hold due to availability issues among the working group. We'll try and get it merged in the next couple of weeks.

**Comment by Pyrdacor** [comment]
Thanks for the update. Hope you will continue soon. If you need a helping hand, just let me know. My time is also very limited but maybe I can find some time slots for the project.

**Comment by Perksey** [comment]
Ah no worries I think we're all good, SuperInvoke I think is ready for Silk.NET and SuperBind is getting ready, so all the stars are starting to align for Preview 4.

File: src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs [hunk 1]
```diff
@@ -873,13 +874,60 @@ public unsafe IMonitor Monitor
...
                 if (monitor == null)
-                    monitor = _glfw.GetPrimaryMonitor();
+                {
+                    if (_windowState != WindowState.Fullscreen)
+                    {
+                        // Determine which monitor this window is on. [6 marks]
+                        var monitors = new GlfwMonitorEnumerable();
+                        foreach (var m in monitors)
+                        {
+                            if (m.Bounds.Contains(Position))
```
Comment (copygirl) [inline src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: This wouldn't return true if the window corner is offscreen. Isn't it better to use the window's center? Even better, when the position isn't on any particular monitor, you test which is closest.

Comment (Perksey) [reply src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: Yeah you're right, I'll change it to use the centre and try and add in an extra test.

Comment (Pyrdacor) [reply src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: Or the most overlap of window and monitor. Like shown here: https://stackoverflow.com/questions/21421074/how-to-create-a-full-screen-window-on-the-current-monitor-with-glfw

Comment (Perksey) [reply src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs]: I've made it use the centre and attempted to come up with a check test, but I didn't like it as it felt overengineered plus I'm not that well-versed with the maths, so I don't feel too confident writing such a solution. I've committed it but only so you could see my potential solution - I immediately reverted it.

**Pull request merged**